internal/provider/kubernetes/controller.go

```diff
@@ -280,7 +330,7 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, request reconcile.
 
 	// Update status for all gateway classes
 	for _, gc := range cc.notAcceptedClasses() {
-		if err := updater(gc, false); err != nil {
+		if err := updater(gc, false, string(status.ReasonOlderGatewayClassExists), status.MsgOlderGatewayClassExists); err != nil {
```

Comment:
the status says `ReasonOlderGatewayClassExists`, how are we so sure that these gateway classes were not accepted because they were younger than the accepted gateway class, maybe they are linked to a different controller ?

Reply:
`notAcceptedClasses()` iterates through all __matched__ GCs and only returns GCs that are not the oldest, e.g. managed, GC.